### PR TITLE
Upgrade to latest build_runner and json_serializable

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: app/build.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  pub_dartlang_org:
+    sources:
+      - 'lib/shared/*.dart'
+      - 'lib/search/backend*.dart'
+    builders:
+      json_serializable:
+        options:
+          require_library_directive: false
+          header: |+
+           // Copyright (c) 2017, the Dart project authors. Please see the AUTHORS file
```
Comment (jakobr-google) [inline app/build.yaml]: 2018?

File: app/pubspec.lock [hunk 2]
```diff
@@ -604,4 +590,4 @@ packages:
     source: hosted
     version: "2.1.13"
 sdks:
-  dart: ">=2.0.0-dev <=2.0.0-dev.15.0"
+  dart: ">=2.0.0-dev <=2.0.0-dev.17.0"
```
Comment (jakobr-google) [inline app/pubspec.lock]: We'll need to update the base image in Dockerfile too, then, right?

File: app/build.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Read about `build.yaml` at https://pub.dartlang.org/packages/build_config
+# To update generated code, run `pub run build_runner build`
+targets:
+  pub_dartlang_org:
+    sources:
+      - 'lib/shared/*.dart'
+      - 'lib/search/backend*.dart'
+    builders:
+      json_serializable:
```
Comment (jakobr-google) [inline app/build.yaml]: So, do `json_serializable` do both `JsonSerializableGenerator` and `JsonLiteralGenerator`?

File: app/pubspec.lock (hunk 2 above)
Comment (isoos) [reply app/pubspec.lock]: I think that can be independent of this upper bound constraint. I had an issue with dev.17 running a local instance, not yet sure what was causing it.

File: app/build.yaml (hunk 3 above)
Comment (kevmoo) [reply app/build.yaml]: I forget – but we don't need the LiteralGenerator any more

File: app/build.yaml (hunk 1 above)
Comment (kevmoo) [reply app/build.yaml]: I left it like this to keep the other files unchanged – when we change something else, we can bump those...

**Pull request merged**